rasa/utils/tensorflow/temp_keras_modules.py

```diff
@@ -13,16 +13,18 @@
 import numpy as np
 import tensorflow as tf
 
-from keras.callbacks import Callback, History
-from keras.engine import training, training_utils, data_adapter, base_layer
-from keras.engine.data_adapter import DataHandler
+from tensorflow.keras.callbacks import Callback, CallbackList, History
+from tensorflow.keras.models import Model
+
+from tensorflow.python.keras.engine import training, training_utils, data_adapter
 from tensorflow.python.eager import context
-from keras.utils import tf_utils, version_utils
+from tensorflow.python.keras.utils import tf_utils, version_utils
```

Comment:
We are making a trade-off here by using a private API of TF, i.e. `tf.python` in favour of keeping `keras` as a dependency. It kind of goes against the recommendation of [TF changelog for version 2.6](https://github.com/tensorflow/tensorflow/releases?page=2)

Are we okay relying on a private API? It might be prone to too many breaking changes between versions? 😬

Reply:
Thanks for pointing this out again! My bad. Not sure what I was thinking. I'll test something and then ping you again.

Reply:
So, to some up what was done before and why:
- what I had wanted to avoid was mixing stuff from `tf.keras` and `keras` and to have a separate keras dependency, which we also had in the poetry file at some point (for I don't know what reason). 
- what I had used as reference was [this](https://github.com/tensorflow/tensorflow/blob/r2.7/tensorflow/python/keras/engine/training.py#L882) (i.e. just looking at updates of the [keras model reference implementation we had used before](https://github.com/tensorflow/tensorflow/blob/v2.3.1/tensorflow/python/keras/engine/training.py#L824-L1146), as was also stated [here](https://github.com/RasaHQ/rasa/blob/main/rasa/utils/tensorflow/temp_keras_modules.py#L25)).

Now we have:
- actually the [tf docs for the tf.keras.Model](https://www.tensorflow.org/api_docs/python/tf/keras/Model) actually link [this](https://github.com/keras-team/keras/blob/v2.7.0/keras/engine/training.py#L66-L2947) 
-  which means the mismatch isn't a potential issue as I thought and we can use `keras` imports directly
- (still don't know why there are those two model implementations, but the `tf.keras` one seems to be the "right" thing)

@dakshvar22 , thanks again!

@usc-m  , could you please double-check the last three changes and the above make sense?

Reply:
So if I'm understanding correctly it looks like `tf.keras` is `keras` everywhere we care about, so there is no mismatch issue and we can drop using the private imports? If that understanding agrees with yours then yes, that all makes sense.

Reply:
.... apart from `tf.keras.Model` which does link to `tf.python.keras` and not to `keras.models.Model`. Would not have expected that from the link in the doc. Am I mixing something up here?

Reply:
If you do it from the REPL by importing the class and `inspect` and doing `inspect.getfile` on the class, it doesn't go to `tf.python` - I guess pycharm is caching the lookups?

Reply:
Yep, just the __file__ attribute also points in the right direction, matching what's in the docs 👍

Reply:
@dakshvar22 and @koernerfelicia , to sum up:  as confirmed by @usc-m (who just figured out the funny tf export story which looks like tf.python.keras should  be unused), the current version (i.e. using `keras` model not the `tf.python.keras` model as reference) should be the way to go. (Thanks again @dakshvar22  and @usc-m :))